src/blockdata/script.rs

```diff
@@ -267,6 +267,14 @@ impl Script {
             .into_script()
     }
 
+    /// Generates OP_RETURN-type of scriptPubkey for a given data
+    pub fn with_op_return(data: &Vec<u8>) -> Self {
```

Comment:
Please take a `&[u8]` rather than an `&Vec<u8>`.

Reply:
fixed with the new commit, as well as the rest of your points